include/picongpu/fields/currentDeposition/Esirkepov/TrajectoryAssignmentShapeFunction.hpp

```diff
@@ -0,0 +1,110 @@
+/* Copyright 2013-2022 Axel Huebl, Heiko Burau, Rene Widera, Sergei Bastrakov
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/fields/currentDeposition/Esirkepov/Line.hpp"
+
+#include <cstdint>
+
+
+namespace picongpu
+{
+    namespace currentSolver
+    {
+        /** Helper for Esirkepov and Esirkepov-like current deposition implementation
```

Comment:
I think this file should be located one level above, as it is used by both `Esirkepov` and `EmZ`. Its namespace also implies that

Reply:
Offline discussed: This file is part of Esirkepov because I see EmZ as an implementation of Esirkepov where parts can be reused. Same as the old `class Base` before. I do not like to move this file one level up because VillaBune is not using this file.